### PR TITLE
ROX-23720: Sanity check and doc improvements for clusterLabels

### PR DESCRIPTION
**Pull request by mclasmeier**
## Description

This is a sub-ticket of [ROX-23467](https://issues.redhat.com/browse/ROX-23467).

This PR:

* improves documentation within the secured-cluster-services Helm chart.
* introduces sanity checks within the Helm templating code to make the feature more robust.

## Checklist
- [ ] Investigated and inspected CI test results
- [x] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

### Here I tell how I validated my change

TODO(replace-me)  
Use this space to explain **how you validated** that **your change functions exactly how you expect it**.
Feel free to attach JSON snippets, curl commands, screenshots, etc. Apply a simple benchmark: would the information you
provided convince any reviewer or any external reader that you did enough to validate your change.

It is acceptable to assume trust and keep this section light, e.g. as a bullet-point list.

It is acceptable to skip testing in cases when CI is sufficient, or it's a markdown or code comment change only.  
It is also acceptable to skip testing for changes that are too taxing to test before merging. In such case you are
responsible for the change after it gets merged which includes reverting, fixing, etc. Make sure you validate the change
ASAP after it gets merged or explain in PR when the validation will be performed.  
Explain here why you skipped testing in case you did so.

Have you created automated tests for your change? Explain here which validation activities you did manually and why so.

### Reminder for reviewers

In addition to reviewing code here, reviewers **must** also review testing and request further testing in case the
performed one does not seem sufficient. As a reviewer, you must not approve the change until you understand the
performed testing and you are satisfied with it.

### CODE REVIEW

File: image/templates/helm/stackrox-secured-cluster/README.md.htpl [hunk 1]
```diff
@@ -238,7 +238,7 @@ chart and their default values:
 |`clusterName`| Name of your cluster. | |
 |`confirmNewClusterName`| You don't need to change this unless you upgrade and change the value for `clusterName`. In this case, set it to the new value of `clusterName`. This option exists to prevent you from [accidentally creating a new cluster with a different name](#changing-cluster-after-deployment). | `null` |
 |`centralEndpoint`| Address of the Central endpoint, including the port number (without a trailing slash). If you are using a non-gRPC capable LoadBalancer, use the WebSocket protocol by prefixing the endpoint address with `wss://`. |`central.stackrox.svc:443` |
-|`clusterLabels`| Custom labels associated with a secured cluster | `{}` |
+|`clusterLabels`| Custom labels associated with a secured cluster. Must be a dictionary of strings. | `{}` |
```
Comment (porridge) [inline image/templates/helm/stackrox-secured-cluster/README.md.htpl:241]:
```suggestion
|`clusterLabels`| Custom labels associated with a secured cluster. Must be a dictionary where both keys and values are strings. | `{}` |
```

File: image/templates/helm/stackrox-secured-cluster/templates/_init.tpl.htpl [hunk 2]
```diff
@@ -66,6 +66,13 @@
 
 {{- include "srox.checkVersionRequirements" $ -}}
 
+{{/* Sanity checks.*/}}
+{{- if not (kindIs "invalid" ._rox.clusterLabels) }}
+  {{- if not (kindIs "map" ._rox.clusterLabels) }}
+    {{ include "srox.fail" (printf "type mismatch: clusterLabels is a %s, but must be a dict" (kindOf ._rox.clusterLabels)) }}
+  {{- end }}
```
Comment (porridge) [inline image/templates/helm/stackrox-secured-cluster/templates/_init.tpl.htpl:73]: Could we maybe iterate over entries and check they strings and not numbers or booleans? 🤔

**Review by porridge — COMMENTED** [review]
Kindly please also make the PR description nicer. 😉

**Comment by porridge** [comment]
I think you need to rebase 🤔

File: pkg/helm/charts/tests/securedclusterservices/testdata/helmtest/base-config.test.yaml
Comment (porridge) [inline pkg/helm/charts/tests/securedclusterservices/testdata/helmtest/base-config.test.yaml:1]: Nit: Maybe add another case where key is a number, for completeness?

**Review by porridge — APPROVED** [review]
Perfect! ❤️

File: pkg/helm/charts/tests/securedclusterservices/testdata/helmtest/base-config.test.yaml
Comment (mclasmeier) [reply pkg/helm/charts/tests/securedclusterservices/testdata/helmtest/base-config.test.yaml:1]: @porridge Sorry for not mentioning this in the PR. I had that initially in the test but then I noticed that the test cannot fail -- for some reason. There must be coercing going on somewhere (wasn't able to quickly locate it) which forces the parsing of numeric keys as strings.

**Pull request merged**